File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    // 1. 각 작업의 배포 가능일 계산
    let daysLeft = progresses.map((progress,index) => Math.ceil((100 - progresses[index])/speeds[index]));
    
    let count = 0;  // 2. 배포될 작업수 카운트
    let maxDay = daysLeft[0];   // 3. 현재까지의 배포된 작업 중 가장 늦은 배포일수
    for(let i=0 ; i < progresses.length ; i++){
        if(maxDay >= daysLeft[i]){  // 4. 배포 가능일이 가장 늦은 배포일보다 빠르다면
            count++;
        }else{  // 5. 배포 예정일이 기준 배포일보다 느리면
            answer.push(count);
            count = 1;
            maxDay = daysLeft[i];
        }
    }
    answer.push(count); // 6. 남은 작업까지 함께 배포
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
import java.io.*;

class Solution {
    
    
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        
        int[] remain = new int[progresses.length];
        for(int i=0; i<progresses.length; i++){
            remain[i] = (int)Math.ceil((100-progresses[i])/(double)speeds[i]);
        }
        
        Queue<Integer> q = new ArrayDeque<>();
        q.offer(remain[0]);
        for(int i=1; i<remain.length; i++){
            if(q.peek() >= remain[i]){
                q.offer(remain[i]);
            }else{
                answer.add(q.size());
                q.clear();
                q.offer(remain[i]);
            }
        }
        
        if(q.size() > 0){
            answer.add(q.size());
        }
        
        // List<Integer> -> int[]
        int[] result = new int[answer.size()];
        for(int i=0 ; i<answer.size(); i++)
            result[i] = answer.get(i);
        
        return result;
    }
}